core/src/telemetry/mod.rs

```diff
@@ -255,18 +232,43 @@ pub fn telemetry_init(opts: &TelemetryOptions) -> Result<&'static GlobalTelemDat
                 ..Default::default()
             };
 
+            let mut console_pretty_layer = None;
+            let mut console_compact_layer = None;
+            let mut forward_layer = None;
+            let mut export_layer = None;
+
             if let Some(ref logger) = opts.logging {
                 match logger {
-                    Logger::Console => {
-                        if opts.tracing.is_none() {
-                            let reg =
-                                tracing_subscriber::registry().with(opts.try_get_env_filter()?);
-                            make_console_logger!(opts, reg);
+                    Logger::Console { filter } => {
+                        // This is silly dupe but can't be avoided without boxing.
+                        if env::var("TEMPORAL_CORE_PRETTY_LOGS").is_ok() {
+                            console_pretty_layer = Some(
+                                tracing_subscriber::fmt::layer()
+                                    .with_target(false)
+                                    .event_format(
+                                        tracing_subscriber::fmt::format()
+                                            .pretty()
+                                            .with_source_location(false),
+                                    )
+                                    .with_filter(EnvFilter::new(filter)),
+                            )
+                        } else {
+                            console_compact_layer = Some(
+                                tracing_subscriber::fmt::layer()
+                                    .with_target(false)
+                                    .event_format(
+                                        tracing_subscriber::fmt::format()
+                                            .compact()
+                                            .with_source_location(false),
+                                    )
+                                    .with_filter(EnvFilter::new(filter)),
+                            )
                         }
                     }
-                    Logger::Forward(filter) => {
-                        log::set_max_level(*filter);
-                        globaldat.core_export_logger = Some(CoreExportLogger::new(*filter));
+                    Logger::Forward { filter } => {
+                        let (export_layer, logs_out) = CoreLogExportLayer::new();
+                        globaldat.logs_out = Some(Mutex::new(logs_out));
```

Comment:
Not new in this PR but is there a way to avoid the mutex here?

Reply:
Nope. Tracing implementations need to access self immutably. The locked operation is very, very cheap.